OloEngine/src/OloEngine/Serialization/MeshBinarySerializer.cpp

```
@@ -0,0 +1,1225 @@
+    // ========================================================================
+
+    namespace
+    {
+        // Stream I/O helpers — accept std::ostream / std::istream so they work
+        // with both file streams and in-memory stringstreams.
+        void WriteBytes(std::ostream& out, const void* data, sizet size)
+        {
+            out.write(reinterpret_cast<const char*>(data), static_cast<std::streamsize>(size));
+        }
+
+        void ReadBytes(std::istream& in, void* data, sizet size)
```

Clang (14.0.6)

[warning] parameter name 'in' is too short, expected at least 3 characters

(readability-identifier-length)